script.js

```diff
@@ -130,117 +130,140 @@ function resetBoxesValues() {
+        this.inToMi = (box1, box2) => box1.value = +(box2.value / 63360).toFixed(6)
+        this.mToCm = (box1, box2) => box1.value = +(box2.value * 100).toFixed(6)
+        this.cmToM = (box1, box2) => box1.value = +(box2.value / 100).toFixed(6)
+        this.mToFt = (box1, box2) => box1.value = +(box2.value * 3.28084).toFixed(6)
+        this.ftToM = (box1, box2) => box1.value = +(box2.value / 3.28084).toFixed(6)
+        this.mToIn = (box1, box2) => box1.value = +(box2.value * 39.3701).toFixed(6)
+        this.inToM = (box1, box2) => box1.value = +(box2.value / 39.3701).toFixed(6)
+        this.cmToFt = (box1, box2) => box1.value = +(box2.value * 0.0328084).toFixed(6)
+        this.ftToCm = (box1, box2) => box1.value = +(box2.value / 0.0328084).toFixed(6)
+        this.cmToIn = (box1, box2) => box1.value = +(box2.value * 0.393701).toFixed(6)
+        this.inToCm = (box1, box2) => box1.value = +(box2.value / 0.3937).toFixed(6)
+        this.ftToIn = (box1, box2) => box1.value = +(box2.value * 12).toFixed(6)
+        this.inToFt = (box1, box2) => box1.value = +(box2.value / 12).toFixed(6)
     }
 }
 
-function conversionChecker(firstMass, secondMass) {
-    if (optionsChildOne.value !== firstMass || optionsChildTwo.value !== secondMass) return false
+function conversionChecker(firstType, secondType) {
+    if (optionsChildOne.value !== firstType || optionsChildTwo.value !== secondType) return false
     else return true
 }
 
+// Consts for Convertion Classes
 const mass = new Mass()
+const length = new Length()
 
 firstBox.addEventListener('keyup', () => {
     // If one of them is empty box
-    if (mass.checker(firstBox, secondBox)) return
-    // KG to LB
-    if (conversionChecker('kg', 'lb')) mass.kgToLb(secondBox, firstBox)
-    // LB to KG
-    else if (conversionChecker('lb', 'kg')) mass.lbToKg(secondBox, firstBox)
-    // KG to GM
-    else if (conversionChecker('kg', 'gm')) mass.kgToGm(secondBox, firstBox)
-    // GM to KG
-    else if (conversionChecker('gm', 'kg')) mass.gmToKg(secondBox, firstBox)
-    // KG to MG
-    else if (conversionChecker('kg', 'mg')) mass.kgToMg(secondBox, firstBox)
-    // MG to KG
-    else if (conversionChecker('mg', 'kg')) mass.mgToKg(secondBox, firstBox)
-    // LB to GM
-    else if (conversionChecker('lb', 'gm')) mass.lbToGm(secondBox, firstBox)
-    // GM to LB
-    else if (conversionChecker('gm', 'lb')) mass.gmToLb(secondBox, firstBox)
-    // LB to MG
-    else if (conversionChecker('lb', 'mg')) mass.lbToMg(secondBox, firstBox)
-    // MG to LB
-    else if (conversionChecker('mg', 'lb')) mass.mgToLb(secondBox, firstBox)
-    // GM to MG
-    else if (conversionChecker('gm', 'mg')) mass.gmToMg(secondBox, firstBox)
-    // MG to GM
-    else if (conversionChecker('mg', 'gm')) mass.mgToGm(secondBox, firstBox)
-})
-
+    if (checker(firstBox, secondBox)) return
+    if (optionsParent.value === 'mass') {
+        // KG to LB
+        if (conversionChecker('kg', 'lb')) mass.kgToLb(secondBox, firstBox)
```

Comment:
instead of checking whether the units are equal to what is in the box why don't we directly utilize what is in the box and call the conversion function directly